src/wallet/api/stake_unlock_result.h

```diff
@@ -0,0 +1,26 @@
+#include "wallet/api/wallet2_api.h"
+#include "wallet/wallet2.h"
+
+#include <string>
+
+
+namespace Wallet {
+
+class WalletImpl;
+class StakeUnlockResultImpl : public StakeUnlockResult
+{
+public:
+    StakeUnlockResultImpl(tools::wallet2::request_stake_unlock_result res);
+    StakeUnlockResultImpl();
+    ~StakeUnlockResultImpl();
+
+    bool success();
+    std::string msg();
+    PendingTransaction * ptx();
+
+private:
+    tools::wallet2::request_stake_unlock_result &result;
```

Comment:
You should store this as a regular value instead of a reference.  The reference will only be valid as long as the original object passed in by reference remains valid, but here you really want the `StakeUnlockResult` object to take over the wrapped response object.